Program.cs

```diff
@@ -1,7 +1,32 @@
-﻿namespace Training {
-    internal class Program {
-        static void Main (string[] args) {
-            Console.WriteLine ("Hello, World!");
-        }
-    }
+﻿using System.Runtime.CompilerServices;
+
+internal class Program {
+   private static void Main (string[] args) {
+      int a = 1, z = 100, guess;
+      Console.Write ($"Guess random number from {a} to {z} and press enter key.");
+      Console.ReadKey ();
+      int attempt = 1;
+      while (true) {
+         guess = (a + z) / 2;
+         char key;
+         Console.Write ($"\nIs {guess} your answer?(Y/N): ");
+         key = Console.ReadKey ().KeyChar;

```

Comment:
For simple applications, like this, it's good to consider case-insensitive answer from users (i.e.,. 'n' | 'N'). Change it accordingly.